### PR TITLE
Raise DataValidationError for malformed responses

### PR DESCRIPTION
**Pull request by felipediel**
## Notice
Closing to submit to the dev branch. We no longer merge into master directly.

## The problem
Sometimes A1 and RM pro return invalid responses for check_sensors(). It happens around 22.9 ºC and 23 ºC, link [here](https://github.com/frankjoke/ioBroker.broadlink2/issues/32#issuecomment-438850921).

The RM pro should send:

```python3
b'\x14\x1e'  # 23.0
```
But it sends:
```python3
b'\xf9\x00'  # 249.0 (-7.0 if signed)
```
The same for A1, but the value is different:
```python3
b'\xf2\x01'  # 242.1 (-13.9 if signed)
```

## Proposed changes
I think we should inform the applications when the data is not reliable because it hinders automation with temperature triggers. I propose to add some filters and raise DataValidationError for malformed responses. I will take the opportunity to allow negative temperature values. This is how the official app works.

- RM and A1 operate between 0 and 50 °C. I propose to accept readings between -5 and 85 °C.
- RM4 uses the HTS2 cable with a working temperature between -10 and 40°C. I propose to accept readings between -25 and 85 °C.
- The relative humidity can't be higher than 100, so let's add this filter too.
- Subclass IntEnum to filter invalid a1 responses (light, air quality and noise).
- Subclass ValueError in related BroadlinkExceptions to provide a flexible and non-verbose interface. The client applications can catch the exceptions specifically (error code) or generically (DataValidationError, ValueError or BroadlinkException).

## Breaking changes
Rename the old DataValidationError to StructureAbnormalError (original name from Broadlink SDK), and create a new DataValidationError class to validate the data in this library. 

Fixes: https://github.com/home-assistant/core/issues/42100

Related issues:
https://github.com/frankjoke/ioBroker.broadlink2/issues/32
https://github.com/lprhodes/homebridge-broadlink-rm/issues/319
https://github.com/home-assistant/core/issues/543

Operating temperature and humidity: http://www.ibroadlink.com/downloads/e-catalog-broadlink.pdf

### CODE REVIEW

File: broadlink/remote.py [hunk 1]
```diff
@@ -80,7 +80,10 @@ def _check_sensors(self, command: int) -> bytes:
         response = self.send_packet(0x6A, packet)
         check_error(response[0x22:0x24])
         payload = self.decrypt(response[0x38:])
-        return bytearray(payload[len(self._request_header) + 4 :])
+        data = payload[len(self._request_header) + 4 :]
+        if data[0] >= 0xc0:  # Firmware issue.
```
Comment (felipediel) [inline broadlink/remote.py:84]: This is the same as:
```python3
if data[0] & 0b11000000 == 0b11000000:
```

**Comment by KTibow** [comment]
Wouldn't it be better to just set it to the actual temperature it should be sending?

File: broadlink/remote.py [hunk 2]
```diff
@@ -80,7 +80,10 @@ def _check_sensors(self, command: int) -> bytes:
...
+            raise exception(-4026, "The device returned malformed data", payload)
+        return data
```
Comment (KTibow) [inline broadlink/remote.py:86]: Why did you remove `bytearray`?

**Review by KTibow — CHANGES_REQUESTED** [review]

**Comment by felipediel** [comment]
> Wouldn't it be better to just set it to the actual temperature it should be sending?

I thought about it, but we don't know if there could be different temperatures that trigger it. It would also be difficult to convert data from A1, which also scrambles data from four other sensors that we cannot guess.

File: broadlink/remote.py (hunk 2 above)
Comment (felipediel) [reply broadlink/remote.py:86]: It is a legacy conversion from Python 2, which indexes bytes and bytearrays differently. We no longer support Python 2, so we don't need it.
